### PR TITLE
options changes only

### PR DESCRIPTION
**Pull request by mohitjaggi**

### CODE REVIEW

File: src/main/scala/com/databricks/spark/csv/CsvParser.scala [hunk 1]
```diff
@@ -16,24 +16,24 @@
...
 /**
  * A collection of static functions for working with CSV files in Spark SQL
  */
 class CsvParser {
-
   private var useHeader: Boolean = false
-  private var delimiter: Character = ','
-  private var quote: Character = '"'
-  private var escape: Character = null
+  private var csvParsingOpts: CSVParsingOpts = CSVParsingOpts()
```
Comment (falaki) [inline src/main/scala/com/databricks/spark/csv/CsvParser.scala]: How come comment, schema, and charset are not included in CSVParsingOpts? I don't clearly see a benefit for this level of indirection yet.

File: src/main/scala/com/databricks/spark/csv/DefaultSource.scala [hunk 2]
```diff
@@ -113,36 +114,52 @@ class DefaultSource
...
-      parseMode,
-      parserLib,
-      ignoreLeadingWhiteSpaceFlag,
-      ignoreTrailingWhiteSpaceFlag,
-      schema,
-      charset,
-      inferSchemaFlag)(sqlContext)
+      useHeader = headerFlag,
+      csvParsingOpts = csvParsingOpts,
+      lineExceptionPolicy = lineParsingOpts,
```
Comment (falaki) [inline src/main/scala/com/databricks/spark/csv/DefaultSource.scala]: What does this additional parameter accomplish that parsing mode does not?

File: src/main/scala/com/databricks/spark/csv/CsvParser.scala (hunk 1 above)
Comment (mohitjaggi) [inline src/main/scala/com/databricks/spark/csv/CsvParser.scala]: as the parser improves a ton of options are being added. it makes sense to organize them in containers for ease of use. imho, schema is a separate thing. i intended csvparsingopts to contain options that are pass to the csv parsing library. i can change the name if there is a better one you can think of

File: src/main/scala/com/databricks/spark/csv/DefaultSource.scala (hunk 2 above)
Comment (mohitjaggi) [inline src/main/scala/com/databricks/spark/csv/DefaultSource.scala]: should be the same. renamed it to indicate that this is a "per line" exception handling policy.

**Comment by HyukjinKwon** [comment]
@mohitjaggi I think some of them are fixed already and the changes are too big without some descriptions to follow. Wouldn't this be ensible if you close this and maybe submit a PR with smaller one? Or rebase this if you are working on this?

**Pull request closed without merging**